.github/workflows/ci.yml

```diff
@@ -80,18 +80,46 @@ jobs:
             hs_err_pid*
             target/surefire-reports/*stream
 
-  linux-arm:
-    name: ARM (${{ matrix.arch }})
+  linux-musl:
     runs-on: ubuntu-20.04 # latest
     strategy:
       matrix:
-        arch: [armv6, armv7, arm64]
+        image:
+          - alpine-3.16-x64
+          - alpine-3.16-x86
+          - alpine-3.16-armv6
+          - alpine-3.16-arm64
     steps:
+      - name: Install qemu/docker
+        run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+        with:
+          submodules: true
       - name: Build ${{ env.PACKAGE_NAME }}
         run: |
-          python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder')"
-          chmod a+x builder
-          ./builder build -p ${{ env.PACKAGE_NAME }} --target=linux-${{ matrix.arch }} --spec=downstream
+          aws s3 cp s3://aws-crt-test-stuff/ci/${{ env.BUILDER_VERSION }}/linux-container-ci.sh ./linux-container-ci.sh && chmod a+x ./linux-container-ci.sh
+          ./linux-container-ci.sh ${{ env.BUILDER_VERSION }} aws-crt-${{ matrix.image }} build -p ${{ env.PACKAGE_NAME }}
+
+  # armv7 needs its own action due to inability to differentiate at runtime armv6 vs armv7 in the jvm:
+  #
+  # At build time we can properly figure out that we're targeting armv7.
+  # At run time we have to force armv7 (via environment variable) in order to achieve proper resource path
+  # resolution.
+  linux-musl-armv7:
+    runs-on: ubuntu-20.04 # latest
+    steps:
+      - name: Install qemu/docker
+        run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Build ${{ env.PACKAGE_NAME }}
+        run: |
+          export AWS_CRT_ARCH=armv7
```

Comment:
just curious, how does this environment variable get all the way into docker ?

Reply:
It didn't, but after the latest builder PR (https://github.com/awslabs/aws-crt-builder/pull/203) it should.  Oops.